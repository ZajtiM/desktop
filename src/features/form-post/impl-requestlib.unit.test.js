const RequestForm = require("./impl-requestlib")

const testOpts = {
    url: 'https://www.strava.com/api/v3/uploads',
    auth: { bearer: '_NOT_A_REAL_BEARER' }
}

const uploadInfo =  {
    file: {
    type: 'file',
    fileName: 'testdata/activity.fit'
    },
    activity_type: 'virtualride',
    data_type: 'tcx',
    name: 'Incyclist Ride',
    description: undefined,
    external_id: '2b498cb0-2241-4cf0-9f77-0279adca2851-1702733150888'
}


describe( 'FormPost Feature: RequestLib Implementation',()=>{

    describe ('createForm',()=>{
        test('normal request',async ()=>{
            const c = new RequestForm()
            const res = await c.createForm(testOpts,uploadInfo)
            
            expect(res).toMatchSnapshot()

        })
    })
}) 