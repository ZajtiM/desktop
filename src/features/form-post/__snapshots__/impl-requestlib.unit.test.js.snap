// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormPost Feature: RequestLib Implementation createForm normal request 1`] = `
{
  "auth": {
    "bearer": "_NOT_A_REAL_BEARER",
  },
  "formData": {
    "activity_type": "virtualride",
    "data_type": "tcx",
    "external_id": "2b498cb0-2241-4cf0-9f77-0279adca2851-1702733150888",
    "file": {
      "options": {
        "filepath": "testdata/activity.fit",
      },
      "value": {
        "data": [
          14,
          32,
          26,
          8,
          14,
          2,
          0,
          0,
          46,
          70,
          73,
          84,
          120,
          188,
          64,
          0,
          1,
          0,
          0,
          5,
          1,
          2,
          132,
          0,
          1,
          0,
          2,
          2,
          132,
          3,
          4,
          140,
          4,
          4,
          134,
          0,
          0,
          15,
          4,
          3,
          232,
          0,
          0,
          48,
          57,
          63,
          224,
          89,
          75,
          64,
          0,
          1,
          0,
          23,
          7,
          0,
          1,
          2,
          2,
          2,
          132,
          4,
          2,
          132,
          27,
          10,
          7,
          3,
          4,
          140,
          5,
          2,
          132,
          253,
          4,
          134,
          0,
          0,
          0,
          255,
          0,
          0,
          73,
          110,
          99,
          121,
          99,
          108,
          105,
          115,
          116,
          0,
          0,
          0,
          0,
          0,
          0,
          100,
          63,
          224,
          89,
          75,
          64,
          0,
          1,
          0,
          207,
          2,
          1,
          16,
          13,
          3,
          1,
          2,
          0,
          1,
          1,
          2,
          3,
          5,
          8,
          13,
          21,
          34,
          55,
          89,
          144,
          233,
          121,
          98,
          219,
          0,
          64,
          0,
          1,
          0,
          20,
          10,
          253,
          4,
          134,
          0,
          4,
          133,
          1,
          4,
          133,
          3,
          1,
          2,
          4,
          1,
          2,
          5,
          4,
          134,
          6,
          2,
          132,
          2,
          2,
          132,
          7,
          2,
          132,
          42,
          1,
          0,
          0,
          63,
          224,
          89,
          76,
          28,
          128,
          103,
          28,
          17,
          13,
          2,
          238,
          152,
          90,
          0,
          0,
          0,
          151,
          9,
          113,
          10,
          18,
          0,
          150,
          2,
          0,
          63,
          224,
          89,
          78,
          28,
          128,
          101,
          85,
          17,
          13,
          3,
          187,
          140,
          90,
          0,
          0,
          2,
          105,
          10,
          226,
          10,
          18,
          0,
          150,
          2,
          0,
          63,
          224,
          89,
          80,
          28,
          128,
          99,
          51,
          17,
          13,
          4,
          182,
          150,
          90,
          0,
          0,
          4,
          137,
          11,
          160,
          10,
          21,
          0,
          150,
          2,
          0,
          63,
          224,
          89,
          82,
          28,
          128,
          96,
          182,
          17,
          13,
          5,
          176,
          158,
          90,
          0,
          0,
          6,
          213,
          12,
          9,
          10,
          21,
          0,
          150,
          2,
          0,
          63,
          224,
          89,
          84,
          28,
          128,
          94,
          102,
          17,
          13,
          6,
          216,
          150,
          90,
          0,
          0,
          9,
          57,
          12,
          64,
          10,
          23,
          0,
          150,
          2,
          0,
          63,
          224,
          89,
          86,
          28,
          128,
          91,
          233,
          17,
          13,
          7,
          233,
          151,
          90,
          0,
          0,
          11,
          170,
          12,
          104,
          10,
          25,
          0,
          150,
          2,
          0,
          63,
          224,
          89,
          88,
          28,
          128,
          89,
          108,
          17,
          13,
          9,
          17,
          152,
          90,
          0,
          0,
          14,
          39,
          12,
          131,
          10,
          25,
          0,
          150,
          2,
          0,
          63,
          224,
          89,
          90,
          28,
          128,
          86,
          238,
          17,
          13,
          10,
          56,
          153,
          90,
          0,
          0,
          16,
          162,
          12,
          166,
          10,
          27,
          0,
          150,
          2,
          0,
          63,
          224,
          89,
          92,
          28,
          128,
          84,
          68,
          17,
          13,
          11,
          119,
          155,
          90,
          0,
          0,
          19,
          47,
          12,
          198,
          10,
          30,
          0,
          150,
          2,
          64,
          0,
          1,
          0,
          18,
          10,
          254,
          2,
          132,
          5,
          1,
          0,
          6,
          1,
          0,
          9,
          4,
          134,
          8,
          4,
          134,
          2,
          4,
          134,
          253,
          4,
          134,
          7,
          4,
          134,
          25,
          2,
          132,
          26,
          2,
          132,
          0,
          0,
          0,
          2,
          58,
          0,
          0,
          19,
          47,
          0,
          0,
          69,
          128,
          63,
          224,
          89,
          75,
          63,
          224,
          89,
          92,
          0,
          0,
          69,
          128,
          0,
          0,
          0,
          1,
          64,
          0,
          1,
          0,
          19,
          6,
          254,
          2,
          132,
          2,
          4,
          134,
          253,
          4,
          134,
          7,
          4,
          134,
          8,
          4,
          134,
          9,
          4,
          134,
          0,
          0,
          0,
          63,
          224,
          89,
          75,
          63,
          224,
          89,
          92,
          0,
          0,
          69,
          128,
          0,
          0,
          69,
          128,
          0,
          0,
          19,
          47,
          64,
          0,
          1,
          0,
          34,
          3,
          253,
          4,
          134,
          1,
          2,
          132,
          0,
          4,
          134,
          0,
          63,
          224,
          89,
          92,
          0,
          1,
          0,
          0,
          69,
          128,
          7,
          30,
        ],
        "type": "Buffer",
      },
    },
    "name": "Incyclist Ride",
  },
  "url": "https://www.strava.com/api/v3/uploads",
}
`;
